lib/note-content-editor.jsx

```diff
@@ -63,6 +93,144 @@ export default class NoteContentEditor extends React.Component {
 		invoke( this, 'editor.focus' );
 	}
 
+	handleReturn = () => {
+		let editorState = this.state.editorState;
+		const key = editorState.getSelection().getFocusKey();
+		const block = editorState.getCurrentContent().getBlockForKey( key );
+		const line = block.getText();
+
+		// matches lines that start with `- ` or `* `
+		// preceded by 0 or more tab characters
+		const listItemRe = /^\t*[-\*]\s/;
+		let listItemMatch;
+
+		if ( line.trim() === '-' || line.trim() === '*' ) {
+			// if current line is just a bullet point and whitesapce,
+			// then clear the line (finishing a list)
+
+			// remove `- ` from the current line
+			editorState = EditorStateModifier.removeRange(
+				editorState,
+				SelectionState.createEmpty( key ).merge( {
+					focusOffset: block.getLength()
+				} )
+			)
```

Comment:
I'm still a fan of creating separate variables here for each step. what if instead of reassigning the `editorState` we created a sequence like this…

```js
const withoutPreviousBullet = removeRange( editorState, … );
const withSelectionOnNewline = forceSelection( withoutPreviousBullet, … );
```